source/common/quic/quic_transport_socket_factory.h

```diff
@@ -114,13 +116,20 @@ class QuicClientTransportSocketFactory : public QuicTransportSocketFactoryBase {
     return fallback_factory_->config();
   }
 
+  std::shared_ptr<quic::QuicCryptoClientConfig> getCryptoConfig();
```

Comment:
nit: Please add a comment here, and explain the behavior of this always returning the same config until the context changes.

Reply:
Can we add a new unit test for this behavior?

Reply:
I'm sure that's true, but it would be nice to have a unit test specifically in for QuicTransportSocketFactory. (Maybe we could move that H3Pool test over?)

Reply:
new test added.